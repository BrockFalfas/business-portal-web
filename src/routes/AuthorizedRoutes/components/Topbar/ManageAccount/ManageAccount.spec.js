import React from 'react';
import { shallow } from 'enzyme';

import { ManageAccount, generateMenuItems } from './index';

const fakeOptions = [
  { key: 'v1', icon: 'plus', label: 'one', className: 'asd' },
  { key: '', icon: 'user', label: 'two', className: 'qwe', action: jest.fn() },
];

const fakeOptionsWithoutKeyAndAction = [
  { icon: 'plus', label: 'one', className: 'asd' },
  { icon: 'user', label: 'two', className: null, action: jest.fn() },
];

const initManageAccount = overrides => {
  const mockProps = {
    logout: jest.fn(),
  };
  const wrapper = shallow(<ManageAccount {...mockProps} {...overrides} />);
  return { mockProps, wrapper };
};

describe('component: <ManageAccount/>', () => {
  describe('renderManageAccount', () => {
    it('should render without crashes', () => {
      const { wrapper } = initManageAccount();
      expect(wrapper).toBeTruthy();
    });
  });

  describe('Menu items generator', () => {
    it('should render without crashes', () => {
      const renderedMenu = generateMenuItems(fakeOptions);
      expect(renderedMenu).toBeTruthy();
    });

    it('should be list of null if there is no action and key', () => {
      const renderedMenu = generateMenuItems(fakeOptionsWithoutKeyAndAction);
      expect(renderedMenu[0]).toBeNull();
    });

    it('handle logout action', () => {
      const { mockProps, wrapper } = initManageAccount();
      const instance = wrapper.instance();

      instance.handleLogout();
      expect(mockProps.logout).toBeCalled();
    });

    it('render dropdown', () => {
      const { wrapper } = initManageAccount();

      expect(wrapper).toMatchSnapshot();
    });
  });
});
