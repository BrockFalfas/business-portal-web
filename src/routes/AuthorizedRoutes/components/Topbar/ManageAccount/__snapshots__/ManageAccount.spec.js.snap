// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: <ManageAccount/> Menu items generator render dropdown 1`] = `
<div
  className="ManageAccount"
>
  <Dropdown
    className="ManageAccount-btn"
    menuClassName="ManageAccount-menu"
    options={
      Array [
        Object {
          "className": "ManageAccount-list-item",
          "key": "/add-team-member",
          "value": <Link
            replace={false}
            to="/add-team-member"
      >
            <Icon
                  type="plus"
            />
             
            <span>
                  Add Team Memebers
            </span>
      </Link>,
        },
        Object {
          "className": "ManageAccount-list-item",
          "key": "/add-invite-contractors",
          "value": <Link
            replace={false}
            to="/add-invite-contractors"
      >
            <Icon
                  type="user"
            />
             
            <span>
                  Add / Invite Contractors
            </span>
      </Link>,
        },
        Object {
          "className": "ManageAccount-list-item",
          "key": "/bank-account",
          "value": <Link
            replace={false}
            to="/bank-account"
      >
            <Icon
                  type="bank"
            />
             
            <span>
                  Bank Account
            </span>
      </Link>,
        },
        Object {
          "className": "ManageAccount-list-item ManageAccount-logout",
          "key": "",
          "value": <div
            onClick={[Function]}
      >
            <Icon
                  type="logout"
            />
             
            <span>
                  Log Out
            </span>
      </div>,
        },
      ]
    }
  >
    <Button
      ghost={true}
      type="primary"
    >
      <Icon
        className="ManageAccount-icon"
        type="setting"
      />
       Manage Account
    </Button>
  </Dropdown>
</div>
`;
